### ISSUE TITLE
Set as desktop background action

### ISSUE THREAD

**Issue by jurplel** [jurplel/qView#309]

**Comment by sonulohani** [comment]
Can i work on this issue.

**Comment by jurplel** [comment]
Sure!

**Comment by sonulohani** [comment]
I am able to set the background in windows. However in Linux it depends on the desktop environment you're using. I have never worked in apple based system, so I've no idea on how to do it.

**Comment by jurplel** [comment]
Mac should be simple: https://davidwalsh.name/desktop-wallpaper-command-line

**Comment by jurplel** [comment]
Linux should use gsettings and qdbus to at least support GNOME and KDE--other desktop environments usually use one of those anyway. There should be a check somewhere when the program starts for if the relevant software is installed, and if not, the set desktop background option should not be visible.

**Comment by Justinzobel** [comment]
Looking forward to this! <3 Qview

**Comment by Phantop** [comment]
What's the status of this? It never got merged into master and it would be very useful if it was!

**Comment by jurplel** [comment]
Well it's not as simple as just merging it, because I have to make two more implementations for it. The linux one in particular is inconsistent, and might not be reasonably possible at all...

**Comment by sonulohani** [comment]
> Well it's not as simple as just merging it, because I have to make two more implementations for it. The linux one in particular is inconsistent, and might not be reasonably possible at all...

I would suggest to merge atleast for windows.

**Comment by jurplel** [comment]
Yeah, that might be what I end up doing. That feature is more common on Windows than elsewhere anyway.

**Comment by timea-techgirl** [comment]
I also need this, want to set some pictures as wallpaper

**Comment by r57zone** [comment]
This is a very useful feature, one of the basic ones, it would be great to see it and replace the standard viewer.

**Comment by jurplel** [comment]
It is worth noting, for the time being you should be able to use "show in explorer" and then right click the image and set it as background there.

**Comment by r57zone** [comment]
The "Set as Desktop" menu option on an image does not work correctly if the image is smaller or larger.
